src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  handleDelete: (todoId: number) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({ todo, handleDelete }) => {
+  const [isHovered, setIsHovered] = useState(false);
+
+  return (
+    <div
+      className={cn('todo', {
+        completed: todo.completed,
+      })}
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
```

Comment:
there is no need to add the logic in react element. try to show/hide remove icon by using CSS

Reply:
Ok, removed the state and events, fixed it with inline styles and relative/absolute positioning.